src/python_testing/TC_DISHM_1_2.py

```diff
@@ -0,0 +1,119 @@
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_DISHM_1_2(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.DishwasherMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    @async_test_body
+    async def test_TC_DISHM_1_2(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        attributes = Clusters.DishwasherMode.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        if self.check_pics("DISHM.S.A0000"):
+            self.print_step(2, "Read SupportedModes attribute")
+            supported_modes = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes))
+
+            asserts.assert_greater_equal(len(supported_modes), 1, "SupportedModes must have at least one entries!")
+
+            supported_modes_dut = []
+            for m in supported_modes:
+                if m.mode in supported_modes_dut:
+                    asserts.fail("SupportedModes must have unique mode values!")
+                else:
+                    supported_modes_dut.append(m.mode)
+
+            labels = []
+            for m in supported_modes:
+                if m.label in labels:
+                    asserts.fail("SupportedModes must have unique mode label values!")
+                else:
+                    labels.append(m.label)
+
+            # common mode tags
+            commonTags = {0x0: 'Auto',
+                          0x1: 'Quick',
+                          0x2: 'Quiet',
+                          0x3: 'LowNoise',
+                          0x4: 'LowEnergy',
+                          0x5: 'Vacation',
+                          0x6: 'Min',
+                          0x7: 'Max',
+                          0x8: 'Night',
+                          0x9: 'Day'}
+
+            runTags = [tag.value for tag in Clusters.DishwasherMode.Enums.ModeTag
+                       if tag is not Clusters.DishwasherMode.Enums.ModeTag.kUnknownEnumValue]
+
+            for m in supported_modes:
+                for t in m.modeTags:
+                    is_mfg = (0x8000 <= t.value and t.value <= 0xBFFF)
```

Comment:
If `is_mfg`, shouldn't we check for the TagName field bits?  Followup OK for this.

Reply:
ok

Reply:
line 89 just below this makes that check.